official/resnet/resnet_run_loop.py

```diff
@@ -266,14 +266,69 @@ def learning_rate_fn(global_step):
                      false_fn=lambda: lr)
     return lr
 
+
+
+  def poly_rate_fn(global_step):
+    """Handles linear scaling rule, gradual warmup, and LR decay.
+
+    The learning rate starts at 0, then it increases linearly per step.  After
+    FLAGS.poly_warmup_epochs, we reach the base learning rate (scaled to account
+    for batch size). The learning rate is then decayed using a polynomial rate
+    decay schedule with power 2.0.
+
+    Args:
+    current_epoch: `Tensor` for current epoch.
+    poly_rate: Polynomial decay rate.
+
+    Returns:
+
+    A scaled `Tensor` for current learning rate.
+    """
+
+    # Learning rate schedule for LARS polynomial schedule
+    if flags.FLAGS.batch_size == 16384:
+      plr = 25.0
+      w_epochs = 5
+    elif flags.FLAGS.batch_size == 32768:
+      plr = 32.0
+      w_epochs = 14
+    elif flags.FLAGS.batch_size == 8192:
+      plr = 10.0
+      w_epochs = 5
+    else:
+      plr = 5.0
+      w_epochs = 5
+
+    w_steps = int(w_epochs * batches_per_epoch)
+    wrate = (
+      plr * tf.cast(global_step, tf.float32) / tf.cast(
+          w_steps, tf.float32))
+
+    num_epochs = 90
```

Comment:
You should be able to grab this from FLAGS

Reply:
Do we need exact batch-sizes?  Is there a formula you can use?  It seems hard to test if we need to hit/trigger exact batch-sizes, meaning we can only test if we have x GPUs running perf GPU batch of z.  Just asking not saying this is wrong.

Reply:
done. Removed exact batch-sizes too.

Reply:
Talked offline, TODO is fine for now.